bugbug/model.py

```diff
@@ -70,26 +69,25 @@ def bugs():
 
         # Split dataset in training and test.
         X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
-
-        if self.undersampling_enabled:
-            # Under-sample the majority classes, as the datasets are imbalanced.
-            X_train, y_train = RandomUnderSampler(random_state=0).fit_sample(X_train, y_train)
-
-        print(f'X_train: {X_train.shape}, y_train: {y_train.shape}')
-        print(f'X_test: {X_test.shape}, y_test: {y_test.shape}')
+        pipeline = make_pipeline(self.sampler, self.clf)
```

Comment:
Why do we need to make a pipeline with imblearn?

Reply:
Making a pipeline makes sure oversampling occurs only on the training data and not on the validation data. Without a pipeline we'll need to oversample before passing the data to cross validate, which will lead to lot of duplicate/similar values ending up in the validation set, thus not giving a correct idea of how well the model is doing.
[https://beckernick.github.io/oversampling-modeling/](url) 
[https://stackoverflow.com/questions/50245684/using-smote-with-gridsearchcv-in-scikit-learn](url).